tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp

```diff
@@ -482,22 +486,26 @@ void check_dense_output(const TimeStepper& stepper,
 
   // Check that the dense output is continuous
   {
-    const Slab slab(0., 1.);
-    Time time = slab.start();
-    double y = 1.;
-    TimeSteppers::History<double, double> history;
-    const auto rhs = [](const double v, const double /*t*/) noexcept {
-      return v;
-    };
-    initialize_history(
-        time, &history, [](const double t) noexcept { return exp(t); }, rhs,
-        slab.duration(), stepper.number_of_past_steps());
-    take_step(&time, &y, &history, stepper, rhs, slab.duration());
-
-    CHECK(get_dense(slab.duration(), std::numeric_limits<double>::epsilon()) ==
-          approx(1.));
-    CHECK(get_dense(slab.duration(),
-                    1. - std::numeric_limits<double>::epsilon()) == approx(y));
+    auto local_approx = approx.epsilon(1e-12);
+    for (const auto time_step :
+         {Slab(0., 1.).duration(), -Slab(-1., 0.).duration()}) {
+      CAPTURE(time_step);
+      Time time = Slab(0., 1.).start().with_slab(time_step.slab());
+      double y = 1.;
+      TimeSteppers::History<double, double> history;
+      const auto rhs = [](const double v, const double /*t*/) noexcept {
+        return v;
+      };
+      initialize_history(
+          time, &history, [](const double t) noexcept { return exp(t); }, rhs,
+          time_step, stepper.number_of_past_steps());
+      take_step(&time, &y, &history, stepper, rhs, time_step);
+
+      CHECK(get_dense(time_step, std::numeric_limits<double>::epsilon() *
```

Comment:
These epsilons are to ensure that the dense output doesn't land directly on a step value, right? Could you add a code comment explaining that?

Reply:
Yes, but it's complicated.  The "correct" test tests 0, eps, 1-eps, and 1, but 0 and 1 (actually I think just 0) are broken in at least one stepper.  I will fix that, but I want this PR to be only changes to the tests, so I can't fix it here.  Do you still want a comment?

Reply:
I think a comment would be good -- I just feel it's helpful  for reading and maintaining the test to understand that the `eps` case actually acts differently in the implementation from the `0` case. I think that's not necessarily obvious without understanding the dense-output mathematics.

Reply:
Good point.  I will add a comment to that effect.